cpp/src/arrow/compute/exec/asof_join_node.cc

```diff
@@ -482,20 +706,22 @@ class CompositeReferenceTable {
 
 class AsofJoinNode : public ExecNode {
   // Advances the RHS as far as possible to be up to date for the current LHS timestamp
-  bool UpdateRhs() {
+  Result<bool> UpdateRhs() {
```

Comment:
What gets change here (When would this raise an error?)

Reply:
This method can fail if `AdvanceAndMemoize` fails, which could happen if the on-key values are unordered.